pkg/adminconfig/evaluator_interfaces.md

```diff
@@ -0,0 +1,135 @@
+Component for Config Policy Decisions
+==========================
+
+## Summary
+
+Design a component that will evaluate config policy decisions based on the fybrikapplication spec and infrastructure (e.g. deployed clusters).
+
+## Goals
+
+1. Design an interface between the config policy evaluator and the manager.
+
+2. Design an interface between the config policy evaluator and OPA
+
+## Interface with the manager
+
+### Input (`adminconfig.EvaluatorInput`)
+
+A dynamic input constructed per a FybrikApplication, per a single dataset.
+It provides general application data such as workload cluster and application properties, as well as dataset details (user requirements, metadata, required actions.
+```
+// WorkloadInfo holds workload details such as cluster/region, type, etc.
+type WorkloadInfo struct {
+	// Cluster where the user workload is running
+	Cluster multicluster.Cluster
+}
+
+// Request is a request to use a specific asset
+type Request struct {
+	// asset identifier
+	DatasetID string
+	// requested interface
+	Interface api.InterfaceDetails
+	// requested usage, e.g. "read": true, "write": false
+	Usage map[api.DataFlow]bool
+}
+
+// EvaluatorInput is an input to Configuration Policies Evaluator.
+// Used to evaluate configuration policies.
+type EvaluatorInput struct {
+	// Workload configuration
+	Workload WorkloadInfo
+	// Application properties
+	AppInfo api.ApplicationDetails
+	// Asset metadata
+	AssetMetadata *assetmetadata.DataDetails
+	// Requirements for asset usage
+	AssetRequirements Request
+	// Governance Actions for reading data (relevant for read scenarios only)
+	GovernanceActions []model.Action
+}
+```
+On top of the dynamic input based on a specific FybrikApplication, an `Infrastructure` object is used by the config policy evaluator to get deployment infrastructure information: available clusters, available storage, bandwidth metrics, etc. Evaluator uses `InfrastructureManager` to obtain the `Infrastructure` data.
+
+In the current implementation, `Infrastructure` is defined as:
+```
+type Infrastructure struct {
+	// Clusters available for deployment
+	Clusters []multicluster.Cluster
+}
+```
+
+### Output (`adminconfig.EvaluatorOutput`)
+
+The result of evaluating config policies on a given dataset contains deployment `Decision` for each capablility, after merging all relevant decisions for the capability.
```

Comment:
can you explain what is the result of "evaluating config policies"?

Reply:
done